### PR TITLE
Initial implementation

### LINKED ISSUE #2: Initial Implementation
## what
* Implement module to provision EKS worker nodes using `terraform-aws-ec2-autoscale-group`

## why
* Terraform managed kubernetes clusters

## references
* https://www.terraform.io/docs/providers/aws/guides/eks-getting-started.html
* https://github.com/cloudposse/terraform-aws-ec2-autoscale-group

## depends on
* https://github.com/cloudposse/terraform-aws-ec2-autoscale-group/issues/1

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Initial implementation of `terraform-aws-eks-workers`

## why
* To provision AWS resources (AutoScaling Group, IAM Role, Security Group) for EKS Workers
* closes #2

### CODE REVIEW

File: README.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  - name: "terraform-aws-ec2-autoscale-group"
+    description: "Terraform module to provision Auto Scaling Group and Launch Template on AWS"
+    url: "https://github.com/cloudposse/terraform-aws-ec2-autoscale-group"
+  - name: "terraform-aws-ec2-instance"
+    description: "Terraform module for providing a general purpose EC2 instance"
+    url: "https://github.com/cloudposse/terraform-aws-ec2-instance"
+  - name: "terraform-aws-ec2-instance-group"
+    description: "Terraform module for provisioning multiple general purpose EC2 hosts for stateful applications"
+    url: "https://github.com/cloudposse/terraform-aws-ec2-instance-group"
+
```
Comment (osterman) [inline README.yaml]: Add our ecs modules

File: README.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+# Short description of this project
+description: |-
+  Terraform module to provision AWS resources to run EC2 worker nodes for [Elastic Container Service for Kubernetes](https://aws.amazon.com/eks/).
+
```
Comment (osterman) [inline README.yaml]: `introduction:`

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,208 @@
+locals {
+  tags = "${merge(var.tags, map("kubernetes.io/cluster/${var.cluster_name}", "owned"))}"
+
+  # userdata for EKS worker nodes to properly configure Kubernetes applications on EC2 instances
+  # https://docs.aws.amazon.com/eks/latest/userguide/launch-workers.html
+  userdata = <<USERDATA
+  #!/bin/bash
+  set -o xtrace
+  /etc/eks/bootstrap.sh --apiserver-endpoint '${var.cluster_endpoint}' --b64-cluster-ca '${var.cluster_certificate_authority_data}' '${var.cluster_name}'
```
Comment (osterman) [inline main.tf]: This script assumes some specific AMIs?

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+  description              = "Allow worker kubelets and pods to receive communication from the cluster control plane"
+  from_port                = 1025
+  to_port                  = 65535
+  protocol                 = "tcp"
+  security_group_id        = "${join("", aws_security_group.default.*.id)}"
+  source_security_group_id = "${var.cluster_security_group_id}"
+  type                     = "ingress"
+}
+
+resource "aws_security_group_rule" "ingress_security_groups" {
```
Comment (osterman) [inline main.tf]: remove `security_groups` suffix

File: main.tf [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+  security_group_id = "${join("", aws_security_group.default.*.id)}"
+  type              = "ingress"
+}
+
+data "aws_ami" "eks_worker" {
+  count = "${var.enabled == "true" ? 1 : 0}"
+
+  filter {
+    name   = "name"
+    values = ["amazon-eks-node-v*"]
```
Comment (osterman) [inline main.tf]: make this a variable so it can be pinned to something more precise.

Comment (osterman) [reply main.tf]: It can default to what you have here.

File: main.tf (hunk 3 above)
Comment (aknysh) [reply main.tf]: yes, EKS AMI
https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-ami.html

File: README.yaml [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+  - name: "terraform-aws-ec2-instance"
+    description: "Terraform module for providing a general purpose EC2 instance"
+    url: "https://github.com/cloudposse/terraform-aws-ec2-instance"
+  - name: "terraform-aws-ec2-instance-group"
+    description: "Terraform module for provisioning multiple general purpose EC2 hosts for stateful applications"
+    url: "https://github.com/cloudposse/terraform-aws-ec2-instance-group"
+
+# Short description of this project
+description: |-
+  Terraform module to provision AWS resources to run EC2 worker nodes for [Elastic Container Service for Kubernetes](https://aws.amazon.com/eks/).
```
Comment (osterman) [inline README.yaml]: Add
> "Instantiate it multiple times to create as many EKS worker node pools as you need and customized to your specific needs."

(or something like this...)

**Pull request merged**